core/src/gl/texture.cpp

```diff
@@ -189,6 +191,8 @@ void Texture::generate(GLuint _textureUnit) {
     GL_CHECK(glTexParameteri(m_target, GL_TEXTURE_WRAP_T, m_options.wrapping.wrapt));
 
     m_generation = RenderState::generation();
+
+    m_mainThreadJobQueue.makeCurrentThreadTarget();
```

Comment:
Still looking at code, so I might be missing something, but should this thread be made current gl one before making calls to gl functions? Can you clarify please.